### PR TITLE
Weirdiness sur la page https://preprod.reconnect.fr/beneficiary/12490/disaffiliate

### PR DESCRIPTION
**Pull request by AntoineAresu**
[Ticket](https://trello.com/c/FbVtMm6D/3660-3-weirdiness-sur-la-page-https-preprodreconnectfr-beneficiary-12490-disaffiliate)

### CODE REVIEW

File: templates/v2/professional/_beneficiaries_list_row.html.twig [hunk 1]
```diff
@@ -19,9 +19,10 @@
         {{ include('v2/professional/_personal_data_row.html.twig') }}
     </td>
     <td class="border-0 h-100 text-center align-middle" {{ stimulus_controller('dropdown-menu') }}>
+        {% set hasOneManageableRelay = app.user.subjectMembre.countManageableRelays(beneficiary) == 1 %}
         {{ include('v2/professional/_action_dropdown.html.twig', {
             removeFromRelay: {
-                path: path('disaffiliate_beneficiary', {'id': beneficiary.id}),
+                path: path(hasOneManageableRelay ? 'disaffiliate_beneficiary' : 'disaffiliate_beneficiary_relay_choice', {'id': beneficiary.id}),
```
Comment (tcheymol) [inline templates/v2/professional/_beneficiaries_list_row.html.twig:25]: Not a huge fan of ternaries inside path function, you loose autocompletion

File: templates/v2/professional/_beneficiaries_list_row.html.twig [hunk 2]
```diff
@@ -19,9 +19,10 @@
         {{ include('v2/professional/_personal_data_row.html.twig') }}
     </td>
     <td class="border-0 h-100 text-center align-middle" {{ stimulus_controller('dropdown-menu') }}>
+        {% set hasOneManageableRelay = app.user.subjectMembre.countManageableRelays(beneficiary) == 1 %}
```
Comment (tcheymol) [inline templates/v2/professional/_beneficiaries_list_row.html.twig:22]: should it be a subject method instead ?

File: src/ControllerV2/BeneficiaryAffiliationController.php [hunk 3]
```diff
@@ -136,13 +127,34 @@ public function disaffiliate(
...
+        if ($relay && $request->isXmlHttpRequest()) {
+            $manager->disaffiliateBeneficiary($beneficiary, $relay);
+
+            return $this->json($beneficiary);
         }
 
         return $this->render('v2/user_affiliation/beneficiary/disaffiliate_beneficiary.html.twig', [
             'beneficiary' => $beneficiary,
-            'relays' => $relays,
+            'relays' => $this->getProfessional()?->getManageableRelays($beneficiary),
```
Comment (tcheymol) [inline src/ControllerV2/BeneficiaryAffiliationController.php:157]: if null, then you have a relays variable with value null. I hope you dont loop on this variable on any template

File: src/ControllerV2/BeneficiaryAffiliationController.php [hunk 4]
```diff
@@ -136,13 +127,34 @@ public function disaffiliate(
                     '%relay%' => $uniqueCommonRelay->getNom(),
                 ]),
             );
+        }
 
-            return $this->redirectToRoute('list_beneficiaries');
+        return $this->redirectToRoute('list_beneficiaries');
```
Comment (tcheymol) [inline src/ControllerV2/BeneficiaryAffiliationController.php:132]: you could also remove all the content of this method with something like
```php
$manageableRelays = $this->getProfessional()?->getManageableRelays($beneficiary)?->first();

return  $manageableRelays->count() === 1 
    ? $this->redirectToRoute('disaffiliate_beneficiary_relay_choice', ['id' => $beneficiqary->getId(), 'relayId' =>  $manageableRelays->first() ?: null])
    : $this->redirectToRoute('list_beneficiaries');
```

File: src/ControllerV2/BeneficiaryAffiliationController.php [hunk 5]
```diff
@@ -136,13 +127,34 @@ public function disaffiliate(
...
+    }
+
+    #[Route(
+        path: '/beneficiary/{id}/disaffiliate/relay/{relayId?}',
```
[the remark this answers is not in the document]
Comment (AntoineAresu) [reply src/ControllerV2/BeneficiaryAffiliationController.php:136]: it was already a nullable param before spliting, this param is used on the same route for ajax call

**Pull request merged**